### PR TITLE
removed the mock blockchain

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -896,9 +894,7 @@ def isopen(self):
 
         return self.proxy.opened.call() != 0
 
-    def deposit(self, our_address, amount):  # pylint: disable=unused-argument
-        """`our_address` is an argument used only in mock_client.py but is also
-        kept here to maintain a consistent interface"""
+    def deposit(self, amount):  # pylint: disable=unused-argument
```
Comment (czepluch) [inline raiden/network/rpc/client.py]: is this pylint comment valid still?

Comment (LefterisJP) [inline raiden/network/rpc/client.py]: I guess not, can be removed

File: raiden/tests/benchmark/profile_transfer.py [hunk 2]
```diff
@@ -183,21 +191,37 @@ def profile_transfer(num_nodes=10, channels_per_node=2):
         for position in range(num_nodes)
     ]
 
-    BlockChainServiceMock.reset()
     blockchain_services = list()
+    tester = tester_state(
```
Comment (LefterisJP) [inline raiden/tests/benchmark/profile_transfer.py]: This code:

```python
tester = tester_state(
        private_keys[0],
        private_keys,
        tester_blockgas_limit(),
    )
    nettingchannel_library_address = tester_nettingchannel_library_address(
        tester_state,
    )
    channelmanager_library_address = tester_channelmanager_library_address(
        tester_state,
        nettingchannel_library_address,
    )
    registry_address = tester_registry_address(
        tester_state,
        channelmanager_library_address,
    )
```

is duplicated here and in `raiden/tests/benchmark/speed.py` and `raiden/tests/benchmark/speed_transfer.py`

I know we would not save too many lines, but can you abstract it into a `tester_and_contracts_init()` function or something with a similar name?

File: raiden/tests/benchmark/speed.py [hunk 3]
```diff
@@ -33,16 +39,32 @@ def setup_apps(amount, tokens, num_transfers, num_nodes, channels_per_node):
         for position in range(num_nodes)
     ]
 
-    BlockChainServiceMock.reset()
     blockchain_services = list()
+    tester = tester_state(
```
Comment (LefterisJP) [inline raiden/tests/benchmark/speed.py]: this is the 1st point of duplication

File: raiden/tests/benchmark/speed_transfer.py [hunk 4]
```diff
@@ -49,24 +55,46 @@ def transfer_speed(num_transfers=100, max_locked=100):  # pylint: disable=too-ma
         for i in range(num_transfers)
     ]
 
-    BlockChainServiceMock._instance = True
-    blockchain_service = BlockChainServiceMock(None, MOCK_REGISTRY_ADDRESS)
-    BlockChainServiceMock._instance = blockchain_service  # pylint: disable=redefined-variable-type
+    blockchain_services = list()
+    tester = tester_state(
```
Comment (LefterisJP) [inline raiden/tests/benchmark/speed_transfer.py]: this is the 2nd point of duplication

File: raiden/tests/fixtures/raiden_network.py [hunk 5]
```diff
@@ -125,4 +125,12 @@ def raiden_network(
 
     _raiden_cleanup(request, raiden_apps)
 
+    # The block_number is primed on the app creation, but after the app is
+    # created all the channels are deployed, for the tester implementation this
+    # will advance the block_number with synchronous execution, making the
```
Comment (LefterisJP) [inline raiden/tests/fixtures/raiden_network.py]: > will advance the block_number with synchronous execution

Question for clarification:

So if we deploy all contracts during tests with the tester blockchain, the blockchain's block number will be advanced with each deployment without giving a chance to the alarm task to fire and as such the raiden app would fall behind.

So for that reason you add the `poll_for_new_block()` here?

File: raiden/tests/unit/test_service.py [hunk 6]
```diff
@@ -27,33 +27,7 @@ def test_ping(raiden_network):
     assert decoded.echo == sha3(ping.encode() + app1.raiden.address)
 
 
-@pytest.mark.timeout(5)
-@pytest.mark.parametrize('blockchain_type', ['mock'])
-@pytest.mark.parametrize('number_of_nodes', [2])
-@pytest.mark.parametrize('transport_class', [UnreliableTransport])
-def test_ping_unreachable(raiden_network):
```
Comment (LefterisJP) [inline raiden/tests/unit/test_service.py]: Why did you remove the `test_ping_unreachable()` test?

File: raiden/raiden_service.py [hunk 7]
```diff
@@ -174,9 +174,12 @@ def __init__(self, chain, private_key_bin, transport, discovery, config):
         alarm.register_callback(lambda _: self.poll_blockchain_events())
         alarm.start()
 
+        # prime the block number cache and set the callbacks
         self._blocknumber = alarm.last_block_number
         alarm.register_callback(self.set_block_number)
 
+        alarm.start()
```
Comment (hackaugusto) [inline raiden/raiden_service.py]: I didn't see that, I must have done a bad job while rebasing, will remove it

File: raiden/tasks.py [hunk 8]
```diff
@@ -176,8 +152,36 @@ def _run(self):  # pylint: disable=method-hidden
...
+
+            self.last_block_number = current_block
+            remove = list()
+            for callback in self.callbacks:
+                try:
+                    result = callback(current_block)
+                except:  # pylint: disable=bare-except
+                    log.exception('unexpected exception on alarm')
+                else:
+                    if result is REMOVE_CALLBACK:
```
Comment (hackaugusto) [inline raiden/tasks.py]: Sure, but that should be done in another pr

File: raiden/tests/benchmark/profile_transfer.py (hunk 2 above)
Comment (hackaugusto) [inline raiden/tests/benchmark/profile_transfer.py]: This should be done in another pr, this one is only removing the mock implementation.

BTW, the benchmark scripts are not working, since they are not executed in the build and have falled behind.

File: raiden/tests/fixtures/raiden_network.py (hunk 5 above)
Comment (hackaugusto) [inline raiden/tests/fixtures/raiden_network.py]: Yes

File: raiden/tests/unit/test_service.py (hunk 6 above)
Comment (hackaugusto) [inline raiden/tests/unit/test_service.py]: I don't see why this test is any good:
- it has an implicit assumption that 5 retries are done
- it has another assumption that these retries will be done in a 2 seconds time span
- It assumes that the other app1 will not send any messages for whatever reason, so it depends on the order of the messages in the list
- It doesn't assert on any useful state of the application afterwards
So, to save me some time, instead of fixing it I removed it
